### PR TITLE
docs: Add agent skills

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

**Comment by stopcoder** [comment]
I validated the create-rule skill with a change created by the agent and one inconsistency between the existing tests and the skill is found:

> Also note the skill recommends using `runLintRulesTests(import.meta.url)` directly in the test file, but the actual function signature requires a string path, and all existing tests use `fileURLToPath(import.meta.url)` first — so our implementation correctly follows the existing codebase convention over the skill's example.

**Review by d3xter666 — COMMENTED** [review]
LGTM

Just one small remark:
Symlinks for `.claude/skills` might not work in some environments.

For the agents plugin we simply copy the skills during the build phase

**Comment by RandomByte** [comment]
> Just one small remark: Symlinks for `.claude/skills` might not work in some environments.
> 
> For the agents plugin we simply copy the skills during the build phase

I saw this practice in other projects, e.g. https://github.com/NVIDIA/OpenShell/blob/main/.claude/skills

Also https://github.com/vercel-labs/skills#installation-methods seems to work with symlinks by default. Let me know if you face any issues with one of your tools!

File: .agents/skills/create-autofix/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+| `FixScope.FullExpression` (0) | Entire expression |
+| `FixScope.FirstChild` (1) | `sap.ui.core.Core.getConfiguration().getTheme` (expression part of call) |
+| `FixScope.SecondChild` (2) | `sap.ui.core.Core.getConfiguration()` |
+| `FixScope.ThirdChild` (3) | `sap.ui.core.Core.getConfiguration` |
+| `FixScope.FourthChild` (4) | `sap.ui.core.Core` |
+
+**Common patterns:**
+- `FullExpression` - When replacing `sap.ui.getCore()` entirely with `Core`
+- `FirstChild` - When replacing `Config.getTheme()` with `Theming.getTheme()` (same args)
+- `SecondChild` - When replacing `Config.setRTL(val)` with `Localization.setRTL(val)` and the Config access needs to be replaced but the method name stays
```
Comment (matz3) [inline .agents/skills/create-autofix/SKILL.md:243]: I don't really see a difference between the two examples here. Maybe the examples can be updated to better show how they differ?

File: .agents/skills/create-autofix/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+  // Test edge cases
+  var result = SomeModule.deprecatedMethod("arg1"); // Return value used
+  SomeModule.deprecatedMethod(); // No arguments
+});
+```
+
+**Run the autofix tests:**
+
+```bash
+npm run unit -- test/lib/autofix/autofix.projects.ts
```
Comment (matz3) [inline .agents/skills/create-autofix/SKILL.md:351]: I think this should be `test/lib/autofix/autofix.fixtures.ts`.

`autofix.projects.ts` runs the fixture projects from `fixtures/linter/projects`, not `fixtures/autofix`.

File: .agents/skills/create-autofix/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+```
+
+**Update snapshots:**
+
+```bash
+npm run unit-update-snapshots -- test/lib/autofix/autofix.projects.ts
```
Comment (matz3) [inline .agents/skills/create-autofix/SKILL.md:357]: Same here, and in the snapshots below.

File: .agents/skills/create-rule/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+```
+test/fixtures/linter/rules/$0/
+  Positive_1/           # Code that SHOULD trigger the rule
+    webapp/
+      Component.js      # (or whatever files are relevant)
+      manifest.json     # (include if needed by the check)
+  Negative_1/           # Code that should NOT trigger the rule
+    webapp/
+      Component.js
+      manifest.json
```
Comment (matz3) [inline .agents/skills/create-rule/SKILL.md:169]: The dir-based testing based on projects currently only works for `AsyncComponentFlags` and `renderer` (see if-clause in `test/lib/linter/_linterHelper.ts#createTestsForFixtures`).

In all other cases, the code will lint each file one by one (isolated).

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
+# CLAUDE.md
```
Comment (matz3) [inline AGENTS.md:1]: AGENTS.md?

File: AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+## Key Resources
+
+- `/resources/api-extract.json` - UI5 API metadata (extracted from SAPUI5 SDK)
+- `/resources/types/pseudo-modules/` - Additional module declarations for pseudo module detection
+- `@sapui5/types` npm package - TypeScript definitions for all SAPUI5 libraries (public/protected API only; does not include `@ui5-restricted` or `@private` API)
+- `docs/Scope-of-Autofix.md` - Lists APIs that cannot be replaced automatically and why
+- `docs/Development.md` - Development guidelines and autofix implementation checklist
+
+## Autofix Development
```
Comment (matz3) [inline AGENTS.md:141]: Refer to the skill here? But as skills are "visible" to agents anyways, maybe this can be completely removed.
